src/lowered.jl

```diff
@@ -391,7 +391,12 @@ function methods_by_execution!(@nospecialize(recurse), methodinfo, docexprs, fra
                 elseif skip_include && (f === modinclude || f === Base.include || f === Core.include)
                     # include calls need to be managed carefully from several standpoints, including
                     # path management and parsing new expressions
-                    add_includes!(methodinfo, mod, @lookup(frame, stmt.args[2]))
+                    mod_or_path = @lookup(frame, stmt.args[2])
+                    if isa(mod_or_path, Module) # include(mod, pathname)
```

Comment:
More directly, can you first check whether `stmt` has 2 or 3 args?

The other issue is that there is another 2-arg invocation of `include`:

```julia
help?> include(f, fname)
  include([mapexpr::Function,] path::AbstractString)

  Evaluate the contents of the input source file in the global scope of the containing module. Every module (except those defined with baremodule) has its own definition of include, which evaluates the
  file in that module. Returns the result of the last evaluated expression of the input file. During including, a task-local include path is set to the directory containing the file. Nested calls to
  include will search relative to that path. This function is typically used to load source interactively, or to combine files in packages that are broken into multiple source files. The argument path
  is normalized using normpath which will resolve relative path tokens such as .. and convert / to the appropriate path separator.

  The optional first argument mapexpr can be used to transform the included code before it is evaluated: for each parsed expression expr in path, the include function actually evaluates mapexpr(expr).
  If it is omitted, mapexpr defaults to identity.

  Use Base.include to evaluate a file into another module.

  │ Julia 1.5
  │
  │  Julia 1.5 is required for passing the mapexpr argument.
```

Perhaps we should distinguish `include` from `Base.include` since in fact these don't seem to be the same function.